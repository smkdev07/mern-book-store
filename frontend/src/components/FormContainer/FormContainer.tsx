import React from 'react';

import { Container, Row, Col } from 'react-bootstrap';

export interface FormContainerProps {
  children: React.ReactNode;
}

const FormContainer: React.FC<FormContainerProps> = ({ children }) => (
  <Container>
    <Row className="justify-content-md-center">
      <Col xs={12} md={6}>
        {children}
      </Col>
    </Row>
  </Container>
);

export default FormContainer;
