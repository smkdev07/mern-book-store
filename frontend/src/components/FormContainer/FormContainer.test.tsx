import React from 'react';
import { shallow } from 'enzyme';

import FormContainer, { FormContainerProps } from './FormContainer';

const setup = (props: FormContainerProps) => {
  return shallow(<FormContainer {...props} />);
};

describe('<FormContainer />', () => {
  it('renders the <FormContainer /> component with children', () => {
    const childElement = <p>Test Element</p>; // use a more realistic test element
    const wrapper = setup({ children: childElement });
    expect(wrapper).toMatchSnapshot();
  });
});
