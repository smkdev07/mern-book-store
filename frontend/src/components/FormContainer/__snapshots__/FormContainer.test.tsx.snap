// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormContainer /> renders the <FormContainer /> component with children 1`] = `ShallowWrapper {}`;
